### ISSUE TITLE
Check for StorageType mismatches in muxLLVMVal

### ISSUE THREAD

**Issue by siddharthist** [GaloisInc/crucible#140]
In the function `muxLLVMVal`, and especially in `muxzero`, the zero value being muxed isn't checked for type equality with the other value. It is probably an error if the type doesn't match. 
https://github.com/GaloisInc/crucible/blob/1546a81bbdc2f3e36d85b85209afe044661a63df/crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Value.hs#L447-L448

**Comment by robdockins** [comment]
Indeed, it looks like not checking if the types match is pure laziness.  It "shouldn't" happen... but you never know.
